### DISCUSSION TITLE
Error 'session not authenticated' after button action 'navigate to' in public form

### DISCUSSION THREAD

**Discussion by GrietVanbillemont** [Budibase/budibase#13206] [Help] [answered]
We have an issue with a button which gives an error 'session not authenticated'. 

At the end of our form, there is a button to send the completed information. There are four actions associated with this button: validate form / save row / trigger automation / navigate to. 

The first three actions succeed, but the fourth action is not performed and then the message 'session not authenticated' appears. The 'navigate to' action should take the user to a screen with a text where the registration (filling out the form) is confirmed. The page itself has 'access' 'public', so access should work for a public user. The 'navigate' action button has as destination 'screen' with the path of the specifiek screen (/bevestiging).

We have several similar apps with a similar button with four actions, where public users do get acces to the page as a fourth button action.

**Comment by ConorWebb96** [comment]
Hey @GrietVanbillemont,

Do you have any permissions attached to the tables you are seeing on the page you are navigating to? Are there any permissions set at the screen level as well?

<img width="833" alt="Screenshot 2024-03-13 at 12 21 09" src="https://github.com/Budibase/budibase/assets/126772285/22eab51b-e057-4988-84bc-e167b3a4fad5">
By default most tables that are created are assigned basic read and write access rights. Can you confirm that you don't have these settings placed on whatever data set you are using on the page you are navigating to?

To get to that modal you need to click the access button that can be seen within the screenshot.

Hopefully this helps.

**Comment by GrietVanbillemont** [comment]
hi @ConorWebb96 

There are no tables on the page were the user is navigating to, only a text with a conformation and instructions. The table were data is recieved from the form has acces write and read 'public'. Also the access to all screens (screen with form and the screen where navigation goes to has access 'public')
The First issue - session not authenticated is occuring in a new app. Since last week we have a second similar issue in an app that has been working fine before, and where no changes have been applied to since the error is occuring. 

The second issue is as followed: At the end of our form, there is a button to send the completed information. There are four actions associated with this button: validate form / save row / trigger automation / navigate to. Untill last week, all four actions were working fine. Since this week, the fourth action is no longer activated (third action automations work) but there is no navigation to a next screen. Now the error 'missing' is appearing. After deleting the fourth action (navigate to), the error 'missing' still appears, step 3 automations still works though.

**Reply by ConorWebb96** [reply]
Is there any chance you could provide an app export that I can have a look through just to see if there are any stray permissions/noticeable issues with the app?

- Are you self-hosted or cloud?
- What version of Budibase are you running?

Thanks in advance for the information.

**Reply by GrietVanbillemont** [reply]
I tried to generate the first issue again, but now we recieve also the error 'missing' instead of 'session not authenticated'. 
So, in both exports, we now experience the second issue, error 'missing' occuring and the fourth action not being proceeded. 

We are running version 2.21.7.
We are self-hosted. Apps are embedded in to our own website:
- https://lichtvis.be/e/registratie (for app Inschrijving Lichtvis)
- https://education.brightbook.io/opleidingen (for app Opleidingen Brightbook)

[Inschrijving Lichtvis-export-1710340897139.tar.gz](https://github.com/Budibase/budibase/files/14589294/Inschrijving.Lichtvis-export-1710340897139.tar.gz)
[Opleidingen Brightbook-export-1710340887061.tar.gz](https://github.com/Budibase/budibase/files/14589296/Opleidingen.Brightbook-export-1710340887061.tar.gz)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @GrietVanbillemont,

Looking at your apps it seems the error message you are receiving is related to the trigger automation action you have attached to the buttons.

These actions don't have any automations attached to them. Therefore when it hits that action it is erroring with the message missing. If you remove these actions the errors won't occur anymore.

<img width="864" alt="Screenshot 2024-03-19 at 11 11 13" src="https://github.com/Budibase/budibase/assets/126772285/87bfd023-1e6a-4eaa-93c8-fa753b9123b4">

Not seeing any authenticated related issues. But it seems you have resolved that on your own.

Thanks for providing the app exports it was helpful.

**Reply by GrietVanbillemont** [reply]
Hi Connor

Two questions related to your answer:

- Why do the automations I created not appear in the drop downlist with automations to be selected? If people fill in the form, the automation 'email na registratie' works fine and people receive a personalized email. If I remove the trigger automation step, will people still receive the email?

- You mention "Not seeing any authenticated related issues. But it seems you have resolved that on your own." I did not resolve it, it did not appear from one day to another :).

**Reply by ConorWebb96** [reply]
Only specific automations can be used here; [App action](https://docs.budibase.com/docs/app-action).
